packages/portfolio-contract/src/portfolio.contract.ts

```diff
@@ -62,20 +61,35 @@ const EVMContractAddressesShape: TypedPattern<EVMContractAddresses> =
     usdc: M.string(),
   });
 
-const EVMContractAddressesMap: TypedPattern<EVMContractAddressesMap> =
-  M.splitRecord(
-    fromEntries(
-      keys(AxelarChain).map(chain => [chain, EVMContractAddressesShape]),
-    ) as Record<AxelarChain, typeof EVMContractAddressesShape>,
-  );
+export type AxelarConfig = {
+  [chain in AxelarChain]: {
+    /**
+     * Axelar chain IDs differ between mainnet and testnet.
+     * See [supported-chains-list.ts](https://github.com/axelarnetwork/axelarjs-sdk/blob/f84c8a21ad9685091002e24cac7001ed1cdac774/src/chains/supported-chains-list.ts)
+     */
+    axelarId: string;
+    contracts: EVMContractAddresses;
+  };
+};
+
+const AxelarChainInfoPattern = M.splitRecord({
+  axelarId: M.string(),
+  contracts: EVMContractAddressesShape,
+});
+
+const AxelarConfigShape: TypedPattern<AxelarConfig> = M.splitRecord(
```

Comment:
`chain-info.build.js` should use this shape.

It should check the shape of all the config info it passes in to *.core.js.